### ISSUE TITLE
loss stuck after certain epoch

### ISSUE THREAD

**Issue by ghost** [Gal4way/TPD#11]
Hi, loss got stuck at 0.012 for last 20 epochs. It was decreasing during wamup steps of lr, but after it lr become constant at base lr and loss is fluctuating around the same level.

I am at 32 epoch now. Should I keep it training in a hope of reducing loss?
or Am I doing something wrong?

Config:
base_learning_rate: 2.44e-7
scheduler_config: 
  # 10000 warmup steps
  target: ldm.lr_scheduler.LambdaLinearScheduler
  params:
    warm_up_steps: [ 10000 ]
    cycle_lengths: [ 10000000000000 ] # incredibly large number to prevent corner cases
    f_start: [ 1e-6 ]
    f_max: [ 1. ]
    f_min: [ 1. ]

![image](https://github.com/user-attachments/assets/14a7984f-3659-4a76-a4b8-3cdc9f519bc3)

**Comment by Gal4way** [comment]
Hi. In my setup, I chose a relatively larger loss value. You might want to adjust it accordingly. Additionally, from my experience, diffusion generative models typically reach a low loss value quickly and then start to fluctuate. However, this does not mean the model is fully trained. It is necessary to continue training, as you will observe improvements in metrics and the quality of generated images even though the loss is still fluctuating. This phenomenon has also been demonstrated in other diffusion generative models.

**Comment by Gal4way** [comment]
The base_learning_rate: 2.44e-7 means that after the warm-up phase, the learning rate is finally adjusted to 2.44e-7. The code of this is inherited from Paint-by-Example and might originally come from Stable Diffusion. Your graph appears to show linear scheduling rather than cosine. If you wish to use cosine scheduling, you may need to adjust this part: target: ldm.lr_scheduler.LambdaLinearScheduler.

**Comment by Gal4way** [comment]
I made a mistake, I mean I chose a relatively larger learning rate.
